ENT-6373 Added diffutils as vendored dependency

craigcomstock opened the pull request:
TODO:
- [x] windows
- [x] linuxes
- [x] aix53 (needed xzcat in PATH hack)
- [x] aix71
- [x] solaris (not 11)
- [x] hpux (was missing xzcat in pkg-build-hpux)
- [x] solaris 11 timed out

Ticket: ENT-6373
Changelog: title

craigcomstock commented:
@cf-bottom why not, let's try this: jenkins with exotics

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=5767)](https://ci.cfengine.com//job/pr-pipeline/5767/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5767/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5767/

Lex-2008 commented:
that's awesome, but I think that to make buildscripts actually build it, you need to add it to the `build-scripts/compile-options` file (`var_append DEPS diff`), like here: https://github.com/cfengine/buildscripts/blob/master/build-scripts/compile-options#L104

craigcomstock commented:
> that's awesome, but I think that to make buildscripts actually build it, you need to add it to the `build-scripts/compile-options` file (`var_append DEPS diffutils`), like here: https://github.com/cfengine/buildscripts/blob/master/build-scripts/compile-options#L104

right. I was doing `install-dependencies diffutils` yesterday. Thanks! :+1:

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=5778)](https://ci.cfengine.com//job/pr-pipeline/5778/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5778/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5778/

craigcomstock commented:
@cf-bottom jenkins with exotics please, thanks.

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=5782)](https://ci.cfengine.com//job/pr-pipeline/5782/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5782/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5782/

craigcomstock commented:
Looks like I've got mingw builds working now. Will retry with exotics.

craigcomstock commented:
:pizza: :tada: diff works on windows and centos6 so far. :)

craigcomstock commented:
As I suspected, solaris and aix (so far) don't immediately support xz compression that the source tarball is provided with.

craigcomstock commented:
Retry solaris and aix71 after installing xz and tar(gtar on solaris) via a prototype policy for infra: https://github.com/mendersoftware/infra/pull/180/files

[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=5798)](https://ci.cfengine.com/job/pr-pipeline/5798/)

craigcomstock commented:
retry exotics after core C fix

[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=5807)](https://ci.cfengine.com/job/pr-pipeline/5807/)

craigcomstock commented on deps-packaging/diffutils/cfbuild-diffutils.spec:
```diff
@@ -0,0 +1,47 @@
+
+%prep
+mkdir -p %{_builddir}
+echo $PATH
```
on aix 53 the version of tar in /opt/freeware/bin isn't even new enough to understand how to use `xz` so likely we will need to "repackage" the tar.xz source package as tar.gz and let the script made by `rpmbuild` proceed with that source file instead so an `if aix53; then do odd stuff; else`

vpodzime reviewed (commented):
Looks good to me. Let's make sure it builds everywhere! :crossed_fingers:

craigcomstock commented:
Started a job for exotics: [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=5819)](https://ci.cfengine.com/job/pr-pipeline/5819/)

and for NON exotics: [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=5820)](https://ci.cfengine.com/job/pr-pipeline/5820/)

craigcomstock commented:
@cf-bottom jenkins with exotics

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=5827)](https://ci.cfengine.com//job/pr-pipeline/5827/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5827/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5827/

cf-bottom reviewed (changes requested):
Please use a company e-mail instead of c***@unreasonablefarm.org

vpodzime reviewed (commented):
Looks good to me. Please squash the commits (with the correct email address).

craigcomstock commented:
hpux is green! :) [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=testing-pr%2Flabel%3DPACKAGES_ia64_hpux_11.23&build=11450)](https://ci.cfengine.com/job/testing-pr/label=PACKAGES_ia64_hpux_11.23/11450/)

craigcomstock commented:
solaris 11 is also green! [![Build Status](https://ci.cfengine.com/buildStatus/icon?job=testing-pr%2Flabel%3DPACKAGES_sparc64_solaris_11&build=11453)](https://ci.cfengine.com/job/testing-pr/label=PACKAGES_sparc64_solaris_11/11453/)

So no, @cf-bottom please do a final round on jenkins with exotics, see how it goes. :) I have squashed commits and fixed the signing email (several commits ago actually) so should be :green_light: after an all green jenkins run.

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=pr-pipeline&build=5849)](https://ci.cfengine.com//job/pr-pipeline/5849/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5849/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5849/

vpodzime commented on deps-packaging/diffutils/mingw/debian/control:
```diff
@@ -0,0 +1,13 @@
+Section: libs
+Architecture: all
+Description: CFEngine Build Automation -- diffutils -- mingw64
+ CFEngine Build Automation -- diffutils -- mingw64
```
Is this duplicated on purpose or just a copy-pasta error?

craigcomstock replied:
Duplicated on most deps so cargo cult copy-pasta recipe. If changed it would tweak lots of debian packages. It won't really matter in the end I think since these are just for the cfbuild-<dep>.deb files that get unpacked and consolidated into cfengine-*.deb.

I tried locally building diffutils deb without the duplication and had no problem running dpkg-deb -I on the result.

I think it is a bit of a pattern for the first line to be a summary: `Description: <summary>` and then subsequent lines indented with a longer description. From looking at a few .deb files I had laying around.

vpodzime reviewed (approved):
Looks good to me. Good job, @craigcomstock :clap:

vpodzime commented:
Merging so that I can test https://github.com/cfengine/core/pull/4353 properly.
